src/protect/addons/assess_car_t_validity.py

```diff
@@ -0,0 +1,121 @@
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import print_function
+from protect.addons.common import TCGAToGTEx
+from protect.common import export_results, get_files_from_filestore, untargz
+from protect.haplotyping.phlat import parse_phlat_file
+
+import os
+import pandas as pd
+
+
+def run_car_t_validity_assessment(job, rsem_files, univ_options, reports_options):
+    """
+    A wrapper for assess_car_t_validity.
+
+    :param dict rsem_files: Results form running rsem
+    :param dict univ_options: Dict of universal options used by almost all tools
+    :param dict reports_options: Options specific to assessing the MHC genes
+    :return: The results of running assess_mhc_genes
+    :rtype: toil.fileStore.FileID
+    """
+    return job.addChildJobFn(assess_car_t_validity, rsem_files['rsem.genes.results'],
+                             univ_options, reports_options).rv()
+
+
+def assess_car_t_validity(job, gene_expression, univ_options, reports_options):
+    """
+    This function creates a report on the available clinical trials and scientific literature
+    available for the overexpressed genes in the specified tumor type.
+    It also gives a list of clinical trials available for other types of cancer with the same
+    overexpressed gene.
+
+    :param dict gene_expression: Results form running rsem
+    :param dict univ_options: Dict of universal options used by almost all tools
+    :param dict reports_options: Options specific to assessing the MHC genes
+    :return: The results of running assess_mhc_genes
+    :rtype: toil.fileStore.FileID
+    """
+    job.fileStore.logToMaster('Running car t validity assessment on %s' % univ_options['patient'])
+    work_dir = os.getcwd()
+    # Take file parameters for both TCGA and GTEX files
+    tumor_type = univ_options['tumor_type']
+
+    input_files = {
+        'rsem_quant.tsv': gene_expression,
+        'car_t_targets.tsv.tar.gz': reports_options['car_t_targets_file']}
+    input_files = get_files_from_filestore(job, input_files, work_dir, docker=False)
+
+    input_files['car_t_targets.tsv'] = untargz(input_files['car_t_targets.tsv.tar.gz'],
+                                                 work_dir)
+
+    target_data = pd.read_table(input_files['car_t_targets.tsv'], index_col=0)
+    patient_df = pd.read_csv('rsem_quant.tsv', sep=' ', delimiter='\t', header='infer', index_col=0)
+    patient_df.index = (patient_df.index).str.replace('\\..*$', '')
+
+    overexpressed = []
+    # See if tumor_type is acceptable
+    try:
+        cancer = TCGAToGTEx[tumor_type]
```

Comment:
`tissue_of_origin` instead of `cancer`. Use the shift + f6 trick to refactor this.